File: solutions/4-data-structures-and-libraries/3171/3171-bfs.py
n, l = [int(i) for i in input().split()]

g = [[] for i in range(n)]

for i in range(l):
    x, y = [int(j) - 1 for j in input().split()]
    g[x] += [y]
    g[y] += [x]

visited = [False] * n

def bfs(root):
    q = [root]
    visited[root] = True
    while len(q) > 0:
        u = q.pop(0)
        for v in g[u]:
            if not visited[v]:
                visited[v] = True
                q += [v]

bfs(0)

print('COMPLETO' if sum(visited) == n else 'INCOMPLETO')

File: solutions/4-data-structures-and-libraries/3171/3171-bfs.cpp
#include <bits/stdc++.h>

using namespace std;

const int maxn = 100; 
vector<int> g[maxn];
bitset<maxn> visited;

void bfs(int root) {
    queue<int> q({ root });
    visited[root] = true;
    while (!q.empty()) {
        int u = q.front();
        q.pop();
        for (int v : g[u]) {
            if (!visited[v]) {
                visited[v] = true;
                q.push(v);
            }
        }
    }
}

int main() {
    int n, l;
    cin >> n >> l;
    for (int i = 0; i < l; ++i) {
        int x, y;
        cin >> x >> y;
        --x; --y;
        g[x].push_back(y);
        g[y].push_back(x);
    }
    bfs(0);
    cout << (visited.count() == n ? "COMPLETO" : "INCOMPLETO") << endl;
    return 0;
}
